exist-core/src/main/antlr/org/exist/xquery/parser/XQueryTree.g

```diff
@@ -970,9 +970,11 @@ throws XPathException
 			    try {
                     QName qn= QName.parse(staticContext, t.getText());
                     int code= Type.getType(qn);
-                    if(!Type.subTypeOf(code, Type.ATOMIC))
-                        throw new XPathException(t, "Type " + qn.toString() + " is not an atomic type");
+                    if (!Type.subTypeOf(code, Type.ATOMIC))
+                        throw new XPathException(t.getLine(), t.getColumn(), ErrorCodes.XPST0051, qn.toString() + " is not atomic");
```

Comment:
Why not use the constructor `public XPathException(Expression expr, ErrorCode errorCode, String errorDesc)`?

Reply:
Haven't tested it, but it could or even should work.

On the other hand there is much more to this specific line:
- is it possible that an ATOMIC_TYPE can _ever_ be a non-atomic/complex type?
  If not we can delete those lines altogether.
- this Exception will be rethrown a few lines after so it would make more sense to look at those... 

and the XQueryTree is littered with `expression.getLine(), expression.getColumn()`
I would propose to open a separate PR where all those use the shorter form maybe even include XQuery.g, if it works out.

Reply:
+1 for PR as it probably makes the error message a bit more clear/focussed.